sympy/matrices/kind.py

```diff
@@ -0,0 +1,97 @@
+# sympy.matrices.kind
+
+from sympy.core.kind import Kind, _NumberKind, NumberKind
+from sympy.core.mul import Mul
+
+
+class MatrixKind(Kind):
+    """
+    Kind for all matrices in SymPy.
+
+    Basic class for this kind is ``MatrixBase`` and ``MatrixExpr``,
+    but any expression representing the matrix can have this.
+
+    Parameters
+    ==========
+
+    element_kind : Kind
+        Kind of the element. Default is
+        :class:`sympy.core.kind.NumberKind`,
+        which means that the matrix contains only numbers.
+
+    Examples
+    ========
+
+    Any instance of matrix class has ``MatrixKind``:
+
+    >>> from sympy import MatrixSymbol
+    >>> A = MatrixSymbol('A', 2,2)
+    >>> A.kind
+    MatrixKind(NumberKind)
+
+    Although expression representing a matrix may be not instance of
+    matrix class, it will have ``MatrixKind`` as well:
```

Comment:
```suggestion
    An expression representing a matrix may not be an instance of
    the Matrix class, but it will be an instance of ``MatrixKind``:
```